File: Leetcode/733. Flood Fill.py
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, color: int) -> List[List[int]]:
        # now i will be using bfs
        r,c=len(image),len(image[0])
        oc=image[sr][sc]
        if color==oc:
            return image
        q=deque( [(sr, sc)] )
        while q:
            i,j=q[0]
            q.popleft()
            if i<0 or j<0 or i>=r or j>=c or image[i][j]!=oc:
                continue
            image[i][j]=color
            #4 n3ighbouts
            q.append((i,j+1))
            q.append((i+1,j))
            q.append((i-1,j))
            q.append((i,j-1))
        return image
        

File: Leetcode/733. Flood Fill.cpp
class Solution {
public:
    // for cpp code i will be using dfs 
    int r,c;
    vector<int> dx={-1,0,1,0};
    vector<int> dy={0,-1,0,1};

    void dfs(int i,int j,int col,int oc,vector<vector<int>>& image){
        //base condition checks
        if(i<0 || j<0 || i>=r || j>=c || image[i][j]!=oc)   return;
        image[i][j]=col;
        for(int k=0;k<4;k++){
            dfs(i+dx[k],j+dy[k],col,oc,image);
        }
    }

    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int color) {
        r=image.size(),c=image[0].size();
        int oc=image[sr][sc];
        if(oc==color)   return image;
        dfs(sr,sc,color,oc,image);
        return image;
    }
};
